docs/DeveloperGuide.md

```diff
@@ -249,20 +249,81 @@ Examples of command use:
+`CodeConnectParser#parseCommand()`.
+3. It is determined the type of command is `mark` and a new `MarkTaskCommandParser` is created and `MarkTaskCommandParser#parse`
+is called, parsing and validating the index.
+4. If index is valid, a new `MarkTaskCommand` is created and returned to `LogicManager`.
+5. `LogicManager` calls `MarkTaskCommand#execute()`. If there exists an unmarked task at index 1, 
+`MarkTaskCommand#createMarkedTask()` is called, creating a copy of the task to be marked with its `status` set to `true`.
+6. `Model#setTask()` is then called to replace the task to be marked with the marked copy.
+
+The following sequence diagram below shows the execution of the `mark` command.
+
+![Interactions Inside the Logic Component for the `mark 1` Command](images/MarkTaskSequenceDiagram.png)
+<div style="text-align: center">Sequence diagram of mark command execution</div>
+
+The activity diagram below summarizes the implementation of a `mark` command. 
 
 ![Activity diagram for execution of a mark command](images/MarkTaskActivityDiagram.png)
  <div style="text-align: center">Activity diagram of mark command execution</div>
 
-![Interactions Inside the Logic Component for the `unmark 1` Command](images/UnmarkTaskSequenceDiagram.png)
-<div style="text-align: center">Sequence diagram of unmark command execution</div>
 
 #### Design considerations
 
-* One design consideration involoved marking/unmarking multiple tasks by adding a space before inputting the index of another task.
+* One design consideration involved marking/unmarking multiple tasks by adding a space before inputting the index of another task.
 However, users could forget to input spaces when inputting indexes of multiple tasks, leading to unintended tasks being
-marked/unmarked. This was considered to be an acceptable trade-off as users would be completing tasks one at a time
-most of the time, so a mass mark/unmark feature is a nice-to-have one.
+marked/unmarked without the users even realising it. Limiting the number of tasks that can be marked/unmarked to 1 is thus 
+considered to be an acceptable trade-off.
+
+### Listing of tasks
+
+#### About
+
+CodeConnect allows users to list all of their tasks with `list` and `list time` commands.
+* `list` command lists all the tasks in the order of most recently added task on top.
+* `list time` command lists all the tasks in the order of the task with the earliest deadline on top.
+
+#### Implementation flow
+
+Listing of tasks is facilitated by `ModelManager`. It contains `filteredTaskList` of type `FilteredList`. 
+It also contains `sortedTaskList` of type `SortedList` that wraps `filteredTaskList` in it. `SortedList` contains the
+method SortedList#setComparator(Comparator<? super E> comparator). The above method is called in 
+`ModelManager#updateSortedTaskList(Comparator<Task> comparator)` that allows for the sorting of `sortedTaskList` with a comparator.
```

Comment:
Perhaps mention somewhere that the `sortedTaskList` is what is finally displayed in the UI?

Reply:
Agreed. Added